src/Lachain.Storage/Repositories/SnapshotIndexRepository.cs

```diff
@@ -87,5 +96,113 @@ private void SetVersion(uint repository, ulong block, ulong version)
                 version.ToBytes()
             );
         }
+
+        private void DeleteVersion(uint repository, ulong block, ulong version)
+        {
+            //Console.WriteLine($"Deleting version {version} for "
+            //    + $"{(RepositoryType) repository} for block {block}");
+            _dbContext.Delete(
+                EntryPrefix.SnapshotIndex.BuildPrefix(
+                    repository.ToBytes().Concat(block.ToBytes()).ToArray()));
+        }
+
+        private void SetDeleteOldSnapshotStatus(byte status)
+        {
+            deleteOldSnapshotStatus = status;
+            var prefix = EntryPrefix.DeleteOldSnapshotStatus.BuildPrefix();
+            var content = new byte[1];
+            content[0] = status;
+            _dbContext.Save(prefix, content);
+        }
+
+        public void DeleteOldSnapshot(ulong depth, ulong totalBlocks)
+        {
+            if (DeleteOldSnapshotStatus <= 1)
+            {
```

Comment:
Can we use some enums instead of using raw integers ?

Reply:
Yes. In that case the condition would be big but much readable.